numpy/lib/nanfunctions.py

```diff
@@ -160,6 +160,8 @@ def _remove_nan_1d(arr1d, overwrite_input=False):
         True if `res` can be modified in place, given the constraint on the
         input
     """
+    if arr1d.dtype == object:
+        return arr1d, True
```

Comment:
A comment on why this is correct would be helpful.

Reply:
The original idea here was to ignore object arrays, thus assuming that (or not bothering with them) they don't contain any `nan`s. But there is apparently a precedent for `nan`-searches in object arrays (xref https://github.com/numpy/numpy/pull/9013), so changed as of https://github.com/numpy/numpy/pull/19821/commits/ecba7133ffef32b817d53bcb2ceeb3113b37bb07.